appium/lib/screen-object/accessors/element.rb

```diff
@@ -104,36 +105,44 @@ def dynamic_text_exists? dynamic_text
       # this is the reason why there is a else condition and a rescue.
       # @param [direction] 'default :down, :up'
       # @return [boolean]
-      def element_visible?(direction = :down)
+      def element_visible?
         default_wait = driver.default_wait
         driver.no_wait
         if exists?
           driver.set_wait(default_wait)
-          scroll(direction)
           true
         else
-          scroll(direction)
           false
         end
-      rescue
-       scroll(direction)
-       false
+      rescue RuntimeError
+        false
       end
 
       # method for scrolling until element is visible.
       # this will NOT return any value.
       # @param [direction] 'Down', 'up'
       def scroll_element_to_view(direction = :down, time_out = 30)
-        wait_until(time_out,'Unable to find element',&->{element_visible?(direction)})
+        wait_until(time_out,'Unable to find element') do
+          return true if element_visible?
+          scroll(direction)
+        end
       end
 
       # method for scrolling until element is visible and click.
       # this will NOT return any value.
       # @param [direction] 'Down', 'up'
-      def scroll_element_to_view_click(direction= :down, time_out = 40)
-        wait_until(time_out,'Unable to find element',&->{element_visible?(direction)})
-        click
+      def scroll_element_to_view_tap(direction= :down, time_out = 40)
+        wait_until(time_out,'Unable to find element') do
+          if element_visible?
+            click
+            true
+          else
+            scroll(direction)
+            false
```

Comment:
Don't return `false` here. Just keep scrolling until `if element_visible?` condition is met and if it doesn't within timeout then method would fail anyways. Currently, when you return false it will fail after 1 scroll